src/daft-plan/src/logical_optimization/optimizer.rs

```diff
@@ -335,6 +335,24 @@ impl Optimizer {
     }
 }
 
+pub struct LogicalOptimizer {}
+
+impl LogicalOptimizer {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl crate::optimizer::Optimizer<LogicalPlan> for LogicalOptimizer {
+    fn batches(&self) -> impl IntoIterator<Item = crate::optimizer::RuleBatch<LogicalPlan>> {
+        vec![crate::optimizer::RuleBatch::new(
+            "Drop Repartition",
+            vec![Box::new(DropRepartition::new())],
+            RuleBatchStrategy::Once,
+        )]
+    }
+}
+
```

Comment:
instead of hardcoding the rules, should we instead have this be part of the optimizer?

```rs
type LogicalRuleBatch = RuleBatch<LogicalPlan>;

pub struct LogicalOptimizer {
  batches: Vec<LogicalRuleBatch>
}

impl Default for LogicalOptimizer {
  fn default() -> Self {
    let batches = vec![/** default batches here */]
    Self {batches}
  }
}
``` 

The reason being -- I know SQL will have some rules that are not relevant in the dataframe api, such as decorrelating subqueries. So we'll eventually need a way to plug in additional rules anyways.

Reply:
For SQL, would we run a set of rules before the logical optimizer, or would we have a different optimizer from the dataframe one? Because if it is the former, I would imagine we could create something like a SQLOptimizer struct to do that. If not, I'll work on making it easier to build optimizer batches

Reply:
We'd likely still want to run all of the normal rules, but plug in a few sql specific ones that run either before or after the normal rule batches. So we could totally run this as a separate optimizer as you suggested.

Reply:
I think what I will do for now is rename Optimizer to be something more generic like RuleExecutor, but keep the LogicalOptimizer implementation the same for now. Once we implement some SQL rules, I think we might have a better picture of how we would want to run those in the logical optimizer, if we choose to do so.